Fix the example DAGs dbt_docs.py and cosmos_seed_dag.py

The `dtb_docs.py` example DAG seems to be misbehaving. This issue was identified as part of  #280.

Detailed stack traces generated by this DAG can be found below. They were obtained by running this task with Tilt, plain Python, and in the CI (Github Actions).

**Acceptance criteria**
* [ ] Tilt can run the DAGs `dtb_docs.py` and `cosmos_seed_dag.py` successfully and there are no stack traces in the task logs
* [ ] Delete the line https://github.com/astronomer/astronomer-cosmos/blob/f447874affe03991a6008a7d4e79c2442e26f8ca/cosmos/tests/test_example_dags.py#L20
* [ ]  The command `hatch run tests.py3.8-2.5:test-integration` runs the `dtb_docs.py` and `cosmos_seed_dag.py` DAGs without issues
* [ ] The CI runs the `dtb_docs.py` and `cosmos_seed_dag.py` DAGs without issues, as part of the jobs `Run-Integration-Tests`

## Exception raised with Tilt

When using `tilt up`, the task `generate_dbt_docs` logs the following:
```
[2023-05-22, 10:44:53 UTC] {adapted_subprocesshook.py:68} INFO - Running command: ['dbt', 'docs', 'generate', '--profile', 'postgres_profile']
[2023-05-22, 10:44:53 UTC] {adapted_subprocesshook.py:79} INFO - Output:
[2023-05-22, 10:44:54 UTC] {adapted_subprocesshook.py:91} INFO - [0m10:44:54  Running with dbt=1.5.0
[2023-05-22, 10:44:54 UTC] {adapted_subprocesshook.py:91} INFO - [0m10:44:54  Encountered an error:
[2023-05-22, 10:44:54 UTC] {adapted_subprocesshook.py:91} INFO - Runtime Error
[2023-05-22, 10:44:54 UTC] {adapted_subprocesshook.py:91} INFO -   No dbt_project.yml found at expected path /tmp/tmpsaevo38a/dbt_project/dbt_project.yml
[2023-05-22, 10:44:54 UTC] {adapted_subprocesshook.py:91} INFO -   Verify that each entry within packages.yml (and their transitive dependencies) contains a file named dbt_project.yml
[2023-05-22, 10:44:54 UTC] {adapted_subprocesshook.py:91} INFO - 
[2023-05-22, 10:44:54 UTC] {adapted_subprocesshook.py:95} INFO - Command exited with return code 2
```
Screenshots:

![Screenshot 2023-05-22 at 12 01 33](https://github.com/astronomer/astronomer-cosmos/assets/272048/55da2d78-fd73-4f4c-894a-af9bc78b46ab)

![Screenshot 2023-05-22 at 11 53 15](https://github.com/astronomer/astronomer-cosmos/assets/272048/59f5edae-2180-4f72-b3dc-530c7ca04def)

## Exception raised locally with plain Python

How to reproduce:
```
export AIRFLOW_CONN_AIRFLOW_DB=postgres://postgres:postgres@0.0.0.0:5433/postgres 
export DBT_ROOT_PATH=`pwd`/dev/dags/dbt
export DBT_DOCS_PATH=`pwd`/dev/dags/dbt-docs
export PYTHONPATH=`pwd`:$PYTHONPATH
export AIRFLOW_HOME=`pwd` 

hatch run airflow init
hatch run tests.py3.8-2.5:test-integration
```

Logs:
```
________________________________ test_example_dag[docs_dag] _________________________________

session = <sqlalchemy.orm.session.Session object at 0x105701940>, dag_id = 'docs_dag'

    @pytest.mark.integration
    @pytest.mark.parametrize("dag_id", get_dag_ids())
    def test_example_dag(session, dag_id: str):
        dag_bag = get_dag_bag()
        dag = dag_bag.get_dag(dag_id)
>       test_utils.run_dag(dag, conn_file_path=EXAMPLE_CONN_FILE)

cosmos/tests/test_example_dags.py:66: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
cosmos/tests/utils.py:24: in run_dag
    return test_dag(dag=dag, conn_file_path=conn_file_path)
../../Library/Application Support/hatch/env/virtual/astronomer-cosmos/cZet8eVR/tests.py3.8-2.5/lib/python3.8/site-packages/airflow/utils/session.py:75: in wrapper
    return func(*args, session=session, **kwargs)
cosmos/tests/utils.py:85: in test_dag
    _run_task(ti, session=session)
cosmos/tests/utils.py:131: in _run_task
    ti._run_raw_task(session=session)
../../Library/Application Support/hatch/env/virtual/astronomer-cosmos/cZet8eVR/tests.py3.8-2.5/lib/python3.8/site-packages/airflow/utils/session.py:72: in wrapper
    return func(*args, **kwargs)
../../Library/Application Support/hatch/env/virtual/astronomer-cosmos/cZet8eVR/tests.py3.8-2.5/lib/python3.8/site-packages/airflow/models/taskinstance.py:1378: in _run_raw_task
    self._execute_task_with_callbacks(context, test_mode)
../../Library/Application Support/hatch/env/virtual/astronomer-cosmos/cZet8eVR/tests.py3.8-2.5/lib/python3.8/site-packages/airflow/models/taskinstance.py:1524: in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
../../Library/Application Support/hatch/env/virtual/astronomer-cosmos/cZet8eVR/tests.py3.8-2.5/lib/python3.8/site-packages/airflow/models/taskinstance.py:1585: in _execute_task
    result = execute_callable(context=context)
cosmos/providers/dbt/core/operators/local.py:361: in execute
    result = self.build_and_run_cmd(context=context)
cosmos/providers/dbt/core/operators/local.py:158: in build_and_run_cmd
    return self.run_command(cmd=dbt_cmd, env=env, context=context)
cosmos/providers/dbt/core/operators/local.py:150: in run_command
    self.callback(tmp_project_dir)
dev/dags/dbt_docs.py:62: in docs_callback
    shutil.copytree(target_dir, DBT_DOCS_PATH)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
(...)
        sys.audit("shutil.copytree", src, dst)
>       with os.scandir(src) as itr:
E       FileNotFoundError: [Errno 2] No such file or directory: '/var/folders/td/522y78v91d1f5wgh67mj3p0m0000gn/T/tmprtmr4ldb/dbt_project/target'

/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/shutil.py:555: FileNotFoundError

```

## Exception raised with CI

Example of Github Actions log:
https://github.com/astronomer/astronomer-cosmos/actions/runs/5044478936/jobs/9047581305

```
ERROR    airflow.models.taskinstance.TaskInstance:taskinstance.py:1889 Task failed with exception
Traceback (most recent call last):
  File "/home/runner/work/astronomer-cosmos/astronomer-cosmos/cosmos/providers/dbt/core/operators/local.py", line 361, in execute
    result = self.build_and_run_cmd(context=context)
  File "/home/runner/work/astronomer-cosmos/astronomer-cosmos/cosmos/providers/dbt/core/operators/local.py", line 158, in build_and_run_cmd
    return self.run_command(cmd=dbt_cmd, env=env, context=context)
  File "/home/runner/work/astronomer-cosmos/astronomer-cosmos/cosmos/providers/dbt/core/operators/local.py", line 150, in run_command
    self.callback(tmp_project_dir)
  File "/home/runner/work/astronomer-cosmos/astronomer-cosmos/dev/dags/dbt_docs.py", line 62, in docs_callback
    shutil.copytree(target_dir, DBT_DOCS_PATH)
  File "/opt/hostedtoolcache/Python/3.9.16/x64/lib/python3.9/shutil.py", line 566, in copytree
    with os.scandir(src) as itr:
FileNotFoundError: [Errno 2] No such file or directory: '/tmp/tmpamvks3p8/dbt_project/target'
```

--- a/cosmos/tests/test_example_dags.py
+++ b/cosmos/tests/test_example_dags.py
@@ -15,7 +15,9 @@ EXAMPLE_DAGS_DIR = Path(__file__).parent.parent.parent / "dev/dags"
 EXAMPLE_CONN_FILE = Path(__file__).parent / "test-connections.yaml"
 AIRFLOW_IGNORE_FILE = EXAMPLE_DAGS_DIR / ".airflowignore"
 
-MIN_VER_DAG_FILE: dict[str, list[str]] = {}
+MIN_VER_DAG_FILE: dict[str, list[str]] = {
+    "2.4": ["cosmos_seed_dag.py"],
+}
 
 # Sort descending based on Versions and convert string to an actual version
 MIN_VER_DAG_FILE_VER: dict[Version, list[str]] = {

--- a/cosmos/tests/test_example_dags.py
+++ b/cosmos/tests/test_example_dags.py
@@ -15,12 +15,7 @@ EXAMPLE_DAGS_DIR = Path(__file__).parent.parent.parent / "dev/dags"
 EXAMPLE_CONN_FILE = Path(__file__).parent / "test-connections.yaml"
 AIRFLOW_IGNORE_FILE = EXAMPLE_DAGS_DIR / ".airflowignore"
 
-MIN_VER_DAG_FILE: dict[str, list[str]] = {
-    "2.4": ["cosmos_seed_dag.py"],
-    "2.7": [
-        "cosmos_seed_dag.py",
-    ],  # Bug: https://github.com/astronomer/astronomer-cosmos/issues/282
-}
+MIN_VER_DAG_FILE: dict[str, list[str]] = {}
 
 # Sort descending based on Versions and convert string to an actual version
 MIN_VER_DAG_FILE_VER: dict[Version, list[str]] = {

--- a/dev/dags/cosmos_seed_dag.py
+++ b/dev/dags/cosmos_seed_dag.py
@@ -20,7 +20,7 @@ from pendulum import datetime
 from cosmos.providers.dbt.core.operators import DbtRunOperationOperator, DbtSeedOperator
 
 DEFAULT_DBT_ROOT_PATH = Path(__file__).parent / "dbt"
-DBT_ROOT_PATH = os.getenv("DBT_ROOT_PATH", DEFAULT_DBT_ROOT_PATH)
+DBT_ROOT_PATH = Path(os.getenv("DBT_ROOT_PATH", DEFAULT_DBT_ROOT_PATH))
 
 with DAG(
     dag_id="extract_dag",
@@ -37,14 +37,14 @@ with DAG(
                 task_id=f"drop_{seed}_if_exists",
                 macro_name="drop_table",
                 args={"table_name": seed},
-                project_dir=DBT_ROOT_PATH,
+                project_dir=DBT_ROOT_PATH / "jaffle_shop",
                 schema="public",
                 conn_id="airflow_db",
             )
 
     jaffle_shop_seed = DbtSeedOperator(
         task_id="seed_jaffle_shop",
-        project_dir=DBT_ROOT_PATH,
+        project_dir=DBT_ROOT_PATH / "jaffle_shop",
         conn_id="airflow_db",
         schema="public",
         outlets=[Dataset("SEED://JAFFLE_SHOP")],

